### DISCUSSION TITLE
View and modify body in NextJs 12 Middlewares

### DISCUSSION THREAD

**Discussion by davidecutrupi** [vercel/next.js#30668] [Help] [answered]
Hi, I would like to use Ajv to verify my API request body.
To do this I would like to use the new Nextjs' feature, Middlewares.
I have read the documents and the reference but have not found how to access the body.

**Comment by gaston-flores** [comment]
I think you want [the API routes middleware example](https://github.com/vercel/next.js/tree/canary/examples/api-routes-middleware). The body of the request should be available in `req.body`.

**Comment by bryanrsmith** [comment]
[`Request` has a `body` property](https://developer.mozilla.org/en-US/docs/Web/API/Request/body), but it seems to always be `null` for me in middleware functions.

If reading the body is not supported in middleware it would be nice for the docs to note that, and maybe for NextRequest to throw a helpful error when the body is accessed. The current behavior is a bit confusing.

**Reply by Juno31** [reply]
in 2023, deeply agree with you

**Comment by neoromantic** [comment]
Same here — can't figure out how to get access to POST body when using middlewares. Any ideas?

**Comment by cj** [comment]
@davidecutrupi  @gaston-flores @bryanrsmith @neoromantic It looks as though this is an intentional feature of Vercel Edge Functions https://github.com/vercel/community/discussions/78#discussioncomment-1937085 but I have suggested they add it in as read-only https://github.com/vercel/community/discussions/78#discussioncomment-1937666

**Reply by dmudro** [reply]
just to avoid confusion I read the initial question as Next.js middleware rather than Vercel Edge functions or Vercel Edge middleware.

**Comment by dmudro** [comment, marked as answer]
@davidecutrupi  a year+ later since posting the question `request.json()` works for me in `v13.1`
